src/borg/repository.py

```diff
@@ -324,9 +324,8 @@ def open_index(self, transaction_id, auto_recover=True):
         index_path = os.path.join(self.path, 'index.%d' % transaction_id).encode('utf-8')
         try:
             return NSIndex.read(index_path)
-        except RuntimeError as error:
-            assert str(error) == 'hashindex_read failed'  # everything else means we're in *deep* trouble
-            logger.warning('Repository index missing or corrupted, trying to recover')
+        except (ValueError, OSError) as exc:
```

Comment:
The diff also suggests that the previous error handling was incomplete, see above:

```py
-                if errno:
-                    PyErr_SetFromErrnoWithFilename(OSError, path)
```

Which was not handled here.